File: Frontend.Angular/src/app/models/UserProfile.ts
export interface UserProfile {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  fullName?: string;
  timeZoneId?: string;
  ipAddress?: string;
  imageUrl?: string;
  deviceId?: string;
  roles: string[];        // multiple roles
  permissions: string[];  // permissions as strings
  exp?: number;           // token expiration (epoch seconds)
}

File: Avancira.BFF/Models/UserProfile.cs
namespace Avancira.BFF.Models;

public class UserProfile
{
    public string Id { get; set; } = default!;
    public string? Email { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? FullName => string.Join(" ", new[] { FirstName, LastName }.Where(s => !string.IsNullOrWhiteSpace(s)));
    public string? ProfileImageUrl { get; set; }
    public string[] Roles { get; set; } = Array.Empty<string>();
}